File: TradingSystem/TradingSystem/inventory/application/store/ProductOrderDTO.cs
namespace TradingSystem.inventory.application.store;

public class ProductOrderDTO
{
    protected long productOderId;
    protected DateTime deliveryDate;
    protected DateTime orderingDate;
    protected List<OrderDTO> orders = new();

    public long ProductOderId
    {
        get => productOderId;
        set => productOderId = value;
    }

    public DateTime DeliveryDate
    {
        get => deliveryDate;
        set => deliveryDate = value;
    }

    public DateTime OrderingDate
    {
        get => orderingDate;
        set => orderingDate = value;
    }

    public List<OrderDTO> Orders
    {
        get => orders;
        set => orders = value ?? throw new ArgumentNullException(nameof(value));
    }

    public override string ToString()
    {
        return $"Id: {productOderId}, orderDate: {orderingDate}, delivDate: {deliveryDate}";
    }
}

File: TradingSystem/WebServerStore/StoreApp/src/app/classes/ProductOrderDTO.ts
import { OrderDTO } from "./OrderDTO";

export class ProductOrderDTO {

    productOrderId: number;
    deliveryDate: Date;
    orderingDate: Date;
    orders: OrderDTO[];

    constructor(
        productOrderId: number,
        deliveryDate: Date,
        orderingDate: Date,
        orders: OrderDTO[]) {
            this.productOrderId = productOrderId;
            this.deliveryDate = deliveryDate;
            this.orderingDate = orderingDate;
            this.orders = orders;
    }
}